src/biogeophys/SnowSnicarMod.F90

```diff
@@ -578,81 +780,235 @@ subroutine SNICAR_RT (flg_snw_ice, bounds, num_nourbanc, filter_nourbanc,  &
+                           enddo
+                           ! piecewise linear interpolate into targeted SNICAR bands in a logscale space
+                           call piecewise_linear_interp1d(16,bcint_wvl_ct,enh_omg_bcint_tmp2,wvl_doint,enh_omg_bcint_intp)
+                           ! update snow single-scattering albedo
+                           enh_omg_bcint_intp2 = 10._r8 ** enh_omg_bcint_intp                           
+                           enh_omg_bcint_intp2 = min(1.0E5_r8, max(enh_omg_bcint_intp2,1._r8)) ! constrain enhancement to a reasonable range
+                           ss_alb_snw_lcl(i)   = 1._r8 - (1._r8 - ss_alb_snw_lcl(i)) * enh_omg_bcint_intp2
+                           ss_alb_snw_lcl(i)   = max(0.5_r8, min(ss_alb_snw_lcl(i),1._r8))
+                           ! reset hydrophilic BC property to 0 since it is accounted by updated snow ss_alb above
+                           ss_alb_aer_lcl(1)       = 0.0
+                           asm_prm_aer_lcl(1)      = 0.0
+                           ext_cff_mss_aer_lcl(1)  = 0.0
+                        endif ! end if BC-snow mixing type
+
+                        ! Dust-snow internal mixing applied to all size bins if activated
+                        ! Dust-snow internal mixing primarily affect snow single-scattering albedo
+                        ! default optics of externally mixed dust at 4 size bins based on effective
+                        ! radius of 1.38um and sigma=2.0 with truncation to each size bin (Flanner et al. 2021 GMD)
+                        ! parameterized dust-snow int mix results based on effective radius of 1.1um and sigma=2.0
+                        ! from (He et al. 2019 JAMES). Thus, the parameterization can be approximately applied to
+                        ! all dust size bins here.
+                        tot_dst_snw_conc = (mss_cnc_aer_lcl(i,5) + mss_cnc_aer_lcl(i,6) + &
+                                            mss_cnc_aer_lcl(i,7) + mss_cnc_aer_lcl(i,8)) * 1.0E6_r8 !kg/kg->ppm
+                        if ( snicar_snodst_intmix .and. (tot_dst_snw_conc > 0._r8) ) then
+                           do idb=1,6
+                              enh_omg_dstint_tmp(idb) = dstint_a1(idb)+dstint_a2(idb)*(tot_dst_snw_conc**dstint_a3(idb))
+                              enh_omg_dstint_tmp2(idb) = LOG10(max(enh_omg_dstint_tmp(idb),1._r8))
+                           enddo
+                           ! piecewise linear interpolate into targeted SNICAR bands in a logscale space
+                           call piecewise_linear_interp1d(6,dstint_wvl_ct,enh_omg_dstint_tmp2,wvl_doint,enh_omg_dstint_intp)
+                           ! update snow single-scattering albedo
+                           enh_omg_dstint_intp2 = 10._r8 ** enh_omg_dstint_intp
+                           enh_omg_dstint_intp2 = min(1.0E5_r8, max(enh_omg_dstint_intp2,1._r8)) ! constrain enhancement to a reasonable range
```

Comment:
Looking at this line more carefully, the "min" part seems redundant because the "max" part cannot be less than 1.

Similarly for enh_omg_bcint_intp2 a few lines up from here.

I will remove the redundant code.